src/mono/mono/mini/aot-compiler.c

```diff
@@ -10030,7 +10030,7 @@ execute_system (const char * command)
 	g_free (wstr);
 
 	g_free (quoted_command);
-#elif defined (HAVE_SYSTEM)
+#elif defined (HAVE_SYSTEM) && !defined(MAC_CATALYST)
```

Comment:
instead of this we should do `set(HAVE_SYSTEM 0)` like we do for iOS in src/mono/cmake/configure.cmake